src/main/java/org/isf/utils/jobjects/PaginatedTableDecorator.java

```diff
@@ -0,0 +1,372 @@
+	 *            - the initial page size
+	 * @param paginatorType
+	 *            - the layout type <code>PAGELINK, PAGEBROWSER, PAGEBROWSER_PAGELINK</code>
+	 * @return {@code PaginatedTableDecorator} object
+	 */
+	public static <T> PaginatedTableDecorator<T> decorate(JTable table, PaginationDataProvider<T> dataProvider, int[] pageSizes, int defaultPageSize,
+					int paginatorType) {
+		PaginatedTableDecorator<T> decorator = new PaginatedTableDecorator<>(table, dataProvider, pageSizes, defaultPageSize, paginatorType);
+		decorator.init();
+		return decorator;
+	}
+
+	/**
+	 * Returns the JPanel with all set (JScrollPane, JTable and PageSelectors)
+	 * @return JPanel
+	 */
+	public JPanel getContentPanel() {
+		return contentPanel;
+	}
+
+	private void init() {
+		initDataModel();
+		initPaginationComponents();
+		initListeners();
+		paginate();
+	}
+
+	private void initListeners() {
+		objectTableModel.addTableModelListener(this::refreshPageButtonPanel);
+		if (table.getRowSorter() != null) {
+			table.getRowSorter().addRowSorterListener(e -> {
+				if (e.getType() == RowSorterEvent.Type.SORT_ORDER_CHANGED) {
+					currentPage = 1;
+					paginate();
+				}
+			});
+		}
+	}
+
+	private void initDataModel() {
+		TableModel model = table.getModel();
+		if (!(model instanceof PageableTableModel)) {
+			throw new IllegalArgumentException("TableModel must be a subclass of PageTableModel");
+		}
+		objectTableModel = (PageableTableModel) model;
+	}
+
+	private void initPaginationComponents() {
+		contentPanel = new JPanel(new BorderLayout());
+		JPanel paginationPanel = createPaginationPanel();
+		contentPanel.add(paginationPanel, BorderLayout.NORTH);
+		contentPanel.add(new JScrollPane(table));
+	}
+
+	private JPanel createPaginationPanel() {
+		JPanel paginationPanel = new JPanel();
+
+		if (paginatorType == PAGEBROWSER || paginatorType == PAGEBROWSER_PAGELINK) {
+
+			paginationPanel.add(getJFirstPageButton());
+			paginationPanel.add(getJPreviousPageButton());
+			paginationPanel.add(getJLabelCurrentPage());
+			paginationPanel.add(getJNextPageButton());
+			paginationPanel.add(getJLastPageButton());
+			paginationPanel.add(Box.createHorizontalStrut(15));
+		}
+
+		if (paginatorType == PAGELINK || paginatorType == PAGEBROWSER_PAGELINK) {
+
+			pageLinkPanel = new JPanel(new GridLayout(1, MaxPagingCompToShow, 3, 3));
+			paginationPanel.add(pageLinkPanel);
+		}
+
+		if (pageSizes != null) {
+			JComboBox<Integer> pageComboBox = new JComboBox<>(Arrays.stream(pageSizes).boxed().toArray(Integer[]::new));
+			pageComboBox.addActionListener((ActionEvent e) -> {
+				// to preserve current rows position
+				int currentPageStartRow = ((currentPage - 1) * currentPageSize) + 1;
+				currentPageSize = (Integer) pageComboBox.getSelectedItem();
+				currentPage = ((currentPageStartRow - 1) / currentPageSize) + 1;
+				paginate();
+			});
+			paginationPanel.add(Box.createHorizontalStrut(15));
+			paginationPanel.add(new JLabel("Page Size: "));
```

Comment:
This literal should be in the properties file shouldn't it?

Reply:
Yep, doing